py-polars/polars/testing/asserts/series.py

```diff
@@ -205,11 +207,11 @@ def _assert_series_values_equal(
     if not unequal.any():
         return
 
-    # Only do inexact checking for numeric types
+    # Only do inexact checking for numeric, non-integer types
     if (
         check_exact
-        or left.dtype not in NUMERIC_DTYPES
-        or right.dtype not in NUMERIC_DTYPES
+        or left.dtype not in NUMERIC_DTYPES.difference(INTEGER_DTYPES)
```

Comment:
🤔 should Decimals be checked for equality?

Reply:
ok, for now, probably not:

https://github.com/pola-rs/polars/blob/6fb626b956f7e7e9fbff0d6ceefb737e2e5e96e1/py-polars/polars/testing/asserts/series.py#L158-L164

Reply:
I think we want tolerant checking to work only for floats. The point of tolerant checking is mostly to account for float imprecision.

The whole point of Decimals is that they are an exact representation of decimal numbers, so doing inexact checking for Decimals makes just as much sense as doing inexact checking for integers.

Even though Decimals are cast to floats at this point, if the values are the same, the resulting float value should be the same. So we can already check them in an exact manner.

```suggestion
        or left.dtype not in FLOAT_DTYPES
```